core/src/main/java/org/hibernate/ogm/cfg/impl/HibernateSearchIntegration.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.cfg.impl;
+
+import java.util.Map;
+import java.util.Properties;
+
+import org.hibernate.cfg.Configuration;
+
+/**
+ * Hibernate Search is an optional dependency, so while we need refer to its constants
+ * to make sure its configured correctly for OGM by default, we don't want to create a
+ * hard dependency. This class should be used as a helper to centralise that.
+ *
+ * @author Sanne Grinovero
+ */
+public final class HibernateSearchIntegration {
+
+	private static final boolean searchIsAvailable = isSearchAvailable();
+	private static String MASSINDEXER_PROPERTY_KEY = null;
```

Comment:
If you see the commit history, in the first commit I just fix the issue w/o refactoring.
The result is the same string constants spread all over the code.

You then have two alternatives:
 - either define the constants somewhere like in org.hibernate.ogm.cfg.impl.InternalProperties - as Strings
 - this solution

I've preferred this one as it's type safe to depend on the actual constant, so the project wouldn't compile in case the changes name in Hibernate Search.
But also, Hibernate Search is not a mandatory dependency: for example Neo4J needs to exclude it.. so there needs to be some special care on making this all trigger only if Search is actually around.